src/main/java/com/kuljava/swiatwsi/services/VillageService.java

```diff
@@ -0,0 +1,34 @@
+package com.kuljava.swiatwsi.services;
+
+import com.kuljava.swiatwsi.world.Village;
+import com.kuljava.swiatwsi.world.VillageRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@AllArgsConstructor
+public class VillageService {
+
+  private final VillageRepository villageRepository;
+  private final FreeCoordinatesFinderService freeCoordinatesFindersService;
+
+  public Optional<Village> saveVillage(String name) {
+    Optional<Village> createdVillage = createVillageWithName(name);
+    createdVillage.ifPresent(villageRepository::save);
+    return Optional.empty();
```

Comment:
You are always returning empty Optional? Weird.

Reply:
Probably should be something like return createVillageWithName(name).map(villageRepository::save).orElse(Optional.empty());